### PR TITLE
Feat/add readme examples

### PR DESCRIPTION
**Pull request by vjdmhd**

### CODE REVIEW

File: examples/sha256/main.go [hunk 1]
```diff
@@ -46,4 +49,44 @@ func main() {
 	}
 	fmt.Printf("merkle proof verify result is %v\n", verifyResult)
 
+	var leavesMmr []types.Leaf
```
Comment (vjdmhd) [inline examples/sha256/main.go:52]: @oshorefueled Please check and make example complete for 256

File: examples/keccak/main.go [hunk 2]
```diff
@@ -46,4 +49,43 @@ func main() {
 	}
 	fmt.Printf("merkle proof verify result is %v\n", verifyResult)
 
+	var leavesMmr []types.Leaf
```
Comment (vjdmhd) [inline examples/keccak/main.go:52]: @oshorefueled Please check and make it to work for keccak

File: mmr/util.go [hunk 3]
```diff
@@ -21,63 +17,3 @@ func (m MemStore) append(pos uint64, elem [][]byte) {
 func (m MemStore) getElem(pos uint64) []byte {
 	return m[pos]
 }
-
```
Comment (vjdmhd) [inline mmr/util.go:24]: @oshorefueled I have removed every unused type and functions from the code, check if anything wrong

File: mmr/mmr.go [hunk 4]
```diff
@@ -131,7 +147,11 @@ func (m *MMR) bagRHSPeaks(rhsPeaks [][]byte) []byte {
 		if lp, rhsPeaks = pop(rhsPeaks); lp == nil {
 			panic("pop")
```
Comment (vjdmhd) [inline mmr/mmr.go:148]: @oshorefueled Please remove every `panic` in the code body, errors should be handled.

File: mmr/util.go (hunk 3 above)
Comment (oshorefueled) [reply mmr/util.go:24]: looks good to me

**Review by seunlanlege — CHANGES_REQUESTED** [review]

**Pull request merged**